src/services/PeerDatabase.ts

```diff
@@ -48,22 +48,20 @@ export default class PeerDatabase {
       if (peer === this.identity.getUri()) {
         continue
       }
-      try {
-        // Check for invalid peer addresses. Validate peer only if not in set.
-        if (!this.peers.has(peer) && shallowValidatePeer(peer)) {
+      // Check for invalid peer addresses. Validate peer only if not in set.
+      if (!this.peers.has(peer) && shallowValidatePeer(peer)) {
+        try {
           const peerInfo = await axios.get(peer + '/info')
           if (peer === peerInfo.data.uri) {
             this.memoryMap.set(peer, peerInfo.data.fullMem)
-            this.peers.add(peer)
+          }
+        } catch (e) {
+          if (process.env.NODE_ENV !== 'test') {
```

Comment:
Can we only add the peer if the info endpoint is a 404? If they return a 500 then there's something wrong no matter what version they're on

Reply:
Sure that's fine. All of this logic is going to be rewritten soon anyway when I get around to moving in the peer-finder module